### PR TITLE
[R4R] CDP rest api automating testing

### PR DESCRIPTION
**Pull request by jmaheswaran**
Automated testing using dredd for the CDP rest api functionality

To test the PR do the following:

`git checkout jm-swagger-rest-automated-testing-cdp`

Then from the `kava` folder run the following to test all the endpoints:

`make test_dredd`

### CODE REVIEW

**Comment by jmaheswaran** [comment]
All 82 tests are now passing and zero failing. PR is ready for review.

To checkout and test the PR do the following:

`git checkout jm-swagger-rest-automated-testing-cdp`

Then from the kava folder run the following to test all the endpoints:

`make test_dredd`

Once it completes building and running you should eventually see 82 tests passing and 0 failing.

**Comment by rhuairahrighairidh** [comment]
I tried running this and got errors. Looks like the setuptest script panics because it can't connect to the rest server. Do you know what's wrong?

**Comment by jmaheswaran** [comment]
> I tried running this and got errors. Looks like the setuptest script panics because it can't connect to the rest server. Do you know what's wrong?

Is something already running on port `1317` ?

**Comment by jmaheswaran** [comment]
You might also need to run `make install` first

File: README.md [hunk 1]
```diff
@@ -19,8 +19,6 @@
 
 ### [Telegram](https://t.me/kavalabs) | [Medium](https://medium.com/kava-labs) | [Validator Chat](https://riot.im/app/#/room/#kava-validators:matrix.org)
 
-### Participate in Kava testnets and [snag a founder badge](./docs/REWARDS.md)!
-
 </div>
 
 Reference implementation of Kava, a blockchain for cross-chain DeFi. Built using the [comsos-sdk](https://github.com/cosmos/cosmos-sdk).
```
Comment (fedekunze) [inline README.md:24]:
```suggestion
Reference implementation of Kava, a blockchain for cross-chain DeFi. Built using the [cosmos-sdk](https://github.com/cosmos/cosmos-sdk).
```

**Comment by karzak** [comment]
Same issue. Ran it on a clean server after `make install` and got a panic:
```
Building go test file, please wait...finished
panic: Get http://localhost:1317/auth/accounts/kava1ffv7nhd3z6sych2qpqkk03ec6hzkmufy0r2s4c: dial tcp 127.0.0.1:1317: connect: connection refused

goroutine 1 [running]:
main.getAccountNumberAndSequenceNumber(0xc0008d5570, 0x106721c, 0x2b, 0x0, 0x0)
        /home/ubuntu/kava/rest_test/setup/setuptest.go:461 +0x349
main.sendMsgToBlockchain(0xc0008d5570, 0x106721c, 0x2b, 0x103a5e6, 0x4, 0x103e77d, 0x8, 0xc00006c000, 0x1, 0x1, ...)
        /home/ubuntu/kava/rest_test/setup/setuptest.go:404 +0x5d
main.sendProposal()
        /home/ubuntu/kava/rest_test/setup/setuptest.go:213 +0x380
main.main()
        /home/ubuntu/kava/rest_test/setup/setuptest.go:46 +0x107
Sending messages to blockchain...finished
```

**Comment by jmaheswaran** [comment]
OK it should be fixed now. I just ran and got zero failing tests and no errors.

File: rest_test/setup/setuptest.go [hunk 2]
```diff
@@ -46,6 +58,126 @@ func main() {
...
+	price, err := sdk.NewDecFromStr("1")
+	if err != nil {
+		panic(err)
+	}
+	// TODO - QUESTION is the price time OK?
+	expiryInt, ok := sdk.NewIntFromString("1925744001")
+	if !ok {
+		panic(ok)
+	}
+	expiry := tmtime.Canonical(time.Unix(expiryInt.Int64(), 0))
```
Comment (karzak) [inline rest_test/setup/setuptest.go:92]: replace lines 87-92 with 
```
expiry := tmtime.Now().Add(time.Second * 100000)
```

File: shortstart.sh [hunk 3]
```diff
@@ -0,0 +1,101 @@
+###########################################################################
```
[the remark this answers is not in the document]
Comment (jmaheswaran) [reply shortstart.sh:1]: 🆗

File: rest_test/setup/setuptest.go (hunk 2 above)
Comment (jmaheswaran) [reply rest_test/setup/setuptest.go:92]: 🆗

**Comment by jmaheswaran** [comment]
Made code review changes from @karzak

**Pull request merged**